File: Box.py
class Box:
    def __init__(self, dimensions, box_id):
        self.dimensions = dimensions
        self.id = box_id
    
    def fits_inside(self, other_box):
        self_dimensions_sorted = sorted(self.dimensions, reverse=True)
        other_dimensions_sorted = sorted(other_box.dimensions, reverse=True)
        
        return all(self_dimensions_sorted[i] < other_dimensions_sorted[i] for i in range(3))
    
    def __repr__(self):
        return f"Box({self.id}, {self.dimensions})"
    
    def __lt__(self, other):
        # Define a comparação para '<' (menor que)
        return self.dimensions < other.dimensions
    
    def __eq__(self, other):
        # Define a comparação para '==' (igual a)
        return self.dimensions == other.dimensions

File: Box.java
import java.util.Arrays;

public class Box implements Comparable<Box> {
    int[] dimensions;
    int box_id;

    public Box(int[] dimensions, int box_id) {
        this.dimensions = dimensions;
        this.box_id = box_id;
        
        Arrays.sort(this.dimensions); // Ensure dimensions are sorted
    }

    public boolean fitsInside(Box other) {
        for (int i = 0; i < 3; i++) {
            if (this.dimensions[i] >= other.dimensions[i]) {
                return false;
            }
        }
        return true;
    }

    public int getBoxId() {
        return box_id;
    }


    @Override
    public int compareTo(Box other) {
        for (int i = 2; i >= 0; i--) { // Compare dimensions in descending order

            if (this.dimensions[i] != other.dimensions[i]) {
                return other.dimensions[i] - this.dimensions[i];
            }
        }
        return 0;
    }
}
